synapse/config/workers.py

```diff
@@ -425,6 +368,50 @@ def _should_this_worker_perform_duty(
         # (By this point, these are either the same value or only one is not None.)
         return bool(new_option_should_run_here or legacy_option_should_run_here)
 
+    def _worker_names_performing_this_duty(
+        self,
+        config: Dict[str, Any],
+        legacy_option_name: str,
+        legacy_app_name: str,
+        modern_instance_map_name: str,
+    ) -> List[str]:
+        """
+        retrieves the names of the workers handling a given duty, by either legacy option or instance_map
+        config: settings read from yaml.
+        legacy_option_name: the old way of enabling options. e.g. 'start_pushers'
+        legacy_app_name: The historical app name. e.g. 'synapse.app.pusher'
+        modern_instance_map_name: the string name of the new instance_map. e.g. 'pusher_instances'
+        """
+
+        legacy_option = config.get(legacy_option_name, True)
+
+        worker_instance_map = config.get(modern_instance_map_name)
+        if worker_instance_map is None:
+            # Default to an empty list, which means "another, unknown, worker is
+            # responsible for it".
+            worker_instance_map = []
+
+            # If no worker instances are set we check if
+            # `legacy_option_name` is set, which means use master
+            if legacy_option:
+                worker_instance_map = ["master"]
+
+            if self.worker_app == legacy_app_name:
+                if legacy_option:
+                    # If we're using `legacy_app_name`, and not using
+                    # `modern_instance_map_name`, then we should have
+                    # explicitly set `legacy_option_name` to false.
+                    raise ConfigError(
+                        f"The '{legacy_option_name}' config option must be disabled in "
+                        "the main synapse process before they can be run in a separate "
+                        "worker.\n"
+                        f"Please add `{legacy_option_name}: false` to the main config.\n",
```

Comment:
This is closer to what we had before?

```suggestion
                        f"The {legacy_option_name} config option must be disabled in the main "
                        "synapse process before they can be run in a separate worker.\n\n"
                        f"Please add ``{legacy_option_name}: false`` to the main config.\n",
```

Reply:
Yeah. When I saw it in the log after, I didn't like how it looked. So I tried to make it match a blend of the *several other styles* used in this one file alone. The double backticks looks horrendous and made my OCD itch. If that's how you think it should be though, I can roll with it

Reply:
The double ticks semes to be used in other messages, leaving the single quotes seems more consistent with the others.

Overall, I don't feel **too* strong about this though.